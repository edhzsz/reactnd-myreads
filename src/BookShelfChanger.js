import React from 'react'

const BookShelfChanger = ({ book, onShelfChange }) => {
  return book
  ? (
    <div className="book-shelf-changer">
      <select value={book.shelf || 'none'} onChange={(event) => onShelfChange(book, event.target.value)}>
        <option value="none" disabled>Move to...</option>
        <option value="currentlyReading">Currently Reading</option>
        <option value="wantToRead">Want to Read</option>
        <option value="read">Read</option>
        {book.shelf && <option value="none">None</option>}
      </select>
    </div>
  ) : null;
}

export default BookShelfChanger;