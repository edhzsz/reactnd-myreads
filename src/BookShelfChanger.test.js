import React from 'react';
import ReactDOM from 'react-dom';
import BookShelfChanger from './BookShelfChanger.js';
import renderer from 'react-test-renderer';

describe('Component: BookShelfChanger', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookShelfChanger />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <BookShelfChanger />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should select the shelf of the book', () => {
    const book = {
      title: "Harry Potter and the Sorcerer's Stone",
      authors: ["J.K. Rowling"],
      imageLinks: {
        smallThumbnail:"http://books.google.com/books/content?id=wrOQLV6xB-wC&printsec=frontcover&img=1&zoom=5&imgtk=AFLRE72G3gA5A-Ka8XjOZGDFLAoUeMQBqZ9y-LCspZ2dzJTugcOcJ4C7FP0tDA8s1h9f480ISXuvYhA_ZpdvRArUL-mZyD4WW7CHyEqHYq9D3kGnrZCNiqxSRhry8TiFDCMWP61ujflB&source=gbs_api",
        thumbnail:"http://books.google.com/books/content?id=wrOQLV6xB-wC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE72G3gA5A-Ka8XjOZGDFLAoUeMQBqZ9y-LCspZ2dzJTugcOcJ4C7FP0tDA8s1h9f480ISXuvYhA_ZpdvRArUL-mZyD4WW7CHyEqHYq9D3kGnrZCNiqxSRhry8TiFDCMWP61ujflB&source=gbs_api",
      },
      shelf: "wantToRead",
      id: "wrOQLV6xB-wC"
    };

    const tree = renderer.create(
      <BookShelfChanger book={book} onShelfChange={() => true} />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});