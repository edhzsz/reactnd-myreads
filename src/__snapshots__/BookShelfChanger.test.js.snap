// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BookShelfChanger should match its empty snapshot 1`] = `null`;

exports[`Component: BookShelfChanger should select the shelf of the book 1`] = `
<div
  className="book-shelf-changer"
>
  <select
    onchange={[Function]}
    value="wantToRead"
  >
    <option
      disabled={true}
      value="none"
    >
      Move to...
    </option>
    <option
      value="currentlyReading"
    >
      Currently Reading
    </option>
    <option
      value="wantToRead"
    >
      Want to Read
    </option>
    <option
      value="read"
    >
      Read
    </option>
    <option
      value="none"
    >
      None
    </option>
  </select>
</div>
`;
